apps/bfd-model/bfd-model-dsl-codegen/bfd-model-dsl-codegen-library/src/main/java/gov/cms/model/dsl/codegen/library/DataTransformer.java

```diff
@@ -0,0 +1,849 @@
+   * in {@code dd-MMM-yyyy HH:mm:ss} format. Valid null values are silently accepted without calling
+   * the Consumer.
+   *
+   * @param fieldName name of the field from which the value originates
+   * @param exists returns true if the value exists
+   * @param value returns the value to copy
+   * @param copier Consumer to receive the timestamp
+   * @return this
+   */
+  public DataTransformer copyOptionalRifTimestamp(
+      String fieldName, BooleanSupplier exists, Supplier<String> value, Consumer<Instant> copier) {
+    if (exists.getAsBoolean()) {
+      return copyRifTimestamp(fieldName, false, value.get(), copier);
+    }
+    return this;
+  }
+
+  /**
+   * Parses the string into a BigDecimal and delivers it to the Consumer. The string value must be a
+   * valid positive or negative numeric format. Valid null values are silently accepted without
+   * calling the Consumer.
+   *
+   * @param fieldName name of the field from which the value originates
+   * @param nullable true if null is a valid value
+   * @param value string containing valid real number
+   * @param copier Consumer to receive the BigDecimal
+   * @return this
+   */
+  public DataTransformer copyAmount(
+      String fieldName, boolean nullable, String value, Consumer<BigDecimal> copier) {
+    if (nonNull(fieldName, value, nullable)) {
+      try {
+        BigDecimal amount = new BigDecimal(value);
+        copier.accept(amount);
+      } catch (NumberFormatException ex) {
+        addError(fieldName, "invalid amount");
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Copies an optional field only if its value exists. Uses lambda expressions for the existence
+   * test as well as the value extraction. Optional fields must be nullable at the database level
+   * but must return non-null values when the supplier is called.
+   *
+   * <p>Parses the string into a BigDecimal and delivers it to the Consumer. The string value must
+   * be a valid positive or negative numeric format. Valid null values are silently accepted without
+   * calling the Consumer.
+   *
+   * @param fieldName name of the field from which the value originates
+   * @param exists returns true if the value exists
+   * @param value returns the value to copy
+   * @param copier Consumer to receive the BigDecimal
+   * @return this
+   */
+  public DataTransformer copyOptionalAmount(
+      String fieldName,
+      BooleanSupplier exists,
+      Supplier<String> value,
+      Consumer<BigDecimal> copier) {
+    if (exists.getAsBoolean()) {
+      return copyAmount(fieldName, false, value.get(), copier);
+    }
+    return this;
+  }
+
+  /**
+   * Checks if the given value is unsigned (positive).
+   *
+   * @param fieldName The name of the attribute the value is associated with (for error tracking).
+   * @param value The value being validated.
+   * @return True if the value is positive (>= 0), alse otherwise.
```

Comment:
typo on the word false. 'non-negative' seems a little better than positive with a qualification here.